Edit documentation to point to the official docs repo (#199)
signalfx/splunk-otel-dotnet: CONTRIBUTING.md

## Documentation

The Splunk Observability documentation is hosted on https://docs.splunk.com/Observability,
which contains all the prescriptive guidance for Splunk Observability products. 
Prescriptive guidance consists of step-by-step instructions, conceptual material,
and decision support for customers. Reference documentation and development 
documentation is hosted on this repository.

> To contribute documentation for this project, open a pull request in the 

Reviewer: "this project" read like this repo to me and got me confused for a bit, perhaps be explicit to get clarity at first reading, something like
```suggestion
To contribute to the Splunk Observability documentation, open a pull request in the 
```
